projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -210,6 +213,603 @@ private static function is_squareish_layout( $attr ) {
+		$target_width = self::get_email_target_width( $rendering_context );
+
+		// Determine layout style and columns from attributes
+		$layout_info = self::get_layout_style_from_attributes( $attr );
+
+		// Build layout content based on style and columns
+		$grid_content = self::build_email_layout_content( $images, $layout_info, $target_width, $email_cell_padding );
+
+		// Use Table_Wrapper_Helper for consistent email rendering
+		$image_table_attrs = array(
+			'style' => 'margin: ' . $email_common_margin . 'px 0; padding: 0; border-collapse: collapse;',
+			'width' => $target_width,
+		);
+
+		$html = \Automattic\WooCommerce\EmailEditor\Integrations\Utils\Table_Wrapper_Helper::render_table_wrapper( $grid_content, $image_table_attrs );
+
+		// Add margin below the block
+		$html .= '<div style="margin-bottom: 2em;"></div>';
+
+		return $html;
+	}
+
+	/**
+	 * Get target width for email rendering.
+	 *
+	 * @param object $rendering_context Email rendering context.
+	 * @return int Target width in pixels.
+	 */
+	private static function get_email_target_width( $rendering_context ) {
+		$target_width = 600; // Default
+
+		if ( ! empty( $rendering_context ) && is_object( $rendering_context ) && method_exists( $rendering_context, 'get_layout_width_without_padding' ) ) {
+			$layout_width_px = $rendering_context->get_layout_width_without_padding();
+			if ( is_string( $layout_width_px ) ) {
+				$parsed_width = \Automattic\WooCommerce\EmailEditor\Integrations\Utils\Styles_Helper::parse_value( $layout_width_px );
+				if ( $parsed_width > 0 ) {
+					$target_width = $parsed_width;
+				}
+			}
+		}
+
+		return $target_width;
+	}
+
+	/**
+	 * Process tiled gallery images for email rendering.
+	 *
+	 * @param array $attr Block attributes containing image data.
+	 * @return array Processed image data for email rendering.
+	 */
+	private static function process_tiled_gallery_images_for_email( $attr ) {
+		$images = array();
+
+		// Determine if this is a squareish layout once
+		$layout_info  = self::get_layout_style_from_attributes( $attr );
+		$is_squareish = self::is_squareish_layout_style( $layout_info['style'] );
+
+		// Get images from IDs (primary data source)
+		if ( ! empty( $attr['ids'] ) && is_array( $attr['ids'] ) ) {
+			foreach ( $attr['ids'] as $id ) {
+				// Validate ID is a positive integer and attachment exists
+				$id = absint( $id );
+				if ( ! $id || ! wp_attachment_is_image( $id ) ) {
+					continue;
+				}
+
+				// For square/circle layouts, get a high-quality square crop
+				if ( $is_squareish ) {
+					// Start with full size image for better quality when resizing
+					$image_url = wp_get_attachment_image_url( $id, 'full' );
+
+					// If we have Photon/Jetpack image processing, request high-quality square crop
+					if ( function_exists( 'jetpack_photon_url' ) && $image_url ) {
+						$image_url = add_query_arg(
+							array(
+								'resize' => '1300,1300', // High-quality square crop for email
```

Comment:
I picked this number based on the image sizes I saw in the current email rendering, but we could change it. The images can get large in certain layouts, e.g. if one is leftover.

300 x 300:
<img width="895" height="718" alt="Screenshot 2025-08-25 at 1 36 21 PM" src="https://github.com/user-attachments/assets/b43a752a-f5e3-49f0-b87c-476bdedd3d04" />

1300 x 1300:
<img width="514" height="599" alt="Screenshot 2025-08-25 at 2 58 04 PM" src="https://github.com/user-attachments/assets/a5b68c73-a1d7-45a4-a3ea-798bbc2cfac3" />